analyze.py

```diff
@@ -143,14 +191,54 @@ def _serialize_json(self, data):
        return json.dumps(data, indent=4)
    
     def _serialize_csv(self, data):
-       return json.dumps(data, indent=3) #fixme - make this export csv instead of json
-
+        with open('output.csv', 'w', newline='') as csvfile:
+            csvwriter = csv.writer(csvfile, delimiter=',',quotechar=',', quoting=csv.QUOTE_NONE,escapechar="\\")	
```

Comment:
are you sure this is the right way to setup the writer? did you read the input.csv file and see those delimiters? did it not use quoting for value borders? i really think this is the part that makes your life a living hell in the student_node section